pkg/controller/user-cluster-controller-manager/resources/reconciler.go

```diff
@@ -1408,6 +1414,35 @@ func (r *reconciler) ensureKubernetesDashboardResourcesAreRemoved(ctx context.Co
 	return nil
 }
 
+func (r *reconciler) ensureOrphanedVMwareCloudDirectorCSIResourcesAreRemoved(ctx context.Context) error {
```

Comment:
Why is this necessary? When re-applying the CSI addon, orphaned resources should automatically be removed, no?

Reply:
I wasn't sure if they are removed automatically. But if they are, we can drop this bit of code.

Reply:
Removed